features/path_rules.feature

```diff
@@ -0,0 +1,72 @@
+@sig-network @conformance @release-1.19
+Feature: Path rules
+
+  An Ingress may define routing rules based on the request path.
+
+  If the HTTP request path matches one of the paths in the
+  Ingress objects, the traffic is routed to its backend service.
+
+  Background:
+    Given a new random namespace
+    Given an Ingress resource
+      """
+      apiVersion: networking.k8s.io/v1
+      kind: Ingress
+      metadata:
+        name: path-rules
+      spec:
+        rules:
+          - host: "path-rules"
+            http:
+              paths:
+                - path: /foo
+                  pathType: Prefix
+                  backend:
+                    serviceName: path-rules-foo
+                    servicePort: 80
+
+                - path: /aaa/bbb/
+                  pathType: Prefix
+                  backend:
+                    serviceName: path-rules-aaa-bbb
+                    servicePort: 80
+
+                - path: /routes with invalid characters are ignored!
+                  pathType: Prefix
+                  backend:
+                    serviceName: ignored
+                    servicePort: 80
+
+                - path: /
+                  pathType: Prefix
+                  backend:
+                    serviceName: path-rules-catchall
+                    servicePort: 80
+
+                - path: /foo
+                  pathType: Exact
+                  backend:
+                    serviceName: path-rules-exact
+                    servicePort: 80
+
+                - path: /bar/
+                  pathType: Exact
+                  backend:
+                    serviceName: path-rules-exact
+                    servicePort: 80
+      """
```

Comment:
@alexgervais maybe another two paths with no `pathType` and one using ImplementationSpecific just to have all the possible cases

Reply:
In `v1beta1`, it is possible to specify an Ingress rule with no `pathType` as it defaults to `ImplementationSpecific`. However, in `v1`, this field has no default value and must be specified (ref. https://github.com/kubernetes/kubernetes/pull/89778). Using a blank `pathType` relates to the discussion about the supported apiVersions in the conformance suite (https://github.com/kubernetes-sigs/ingress-controller-conformance/issues/40)

I will update this sample to reflect the changes from `serviceName` to `service.name` and `servicePort` to `service.port` as it was an oversight.

As for `ImplementationSpecific`, it would make sense to test, as it is part of the API, but I'm wondering how we might assert a behavior that's unique to each implementation 🤔